src/schedule/zephyr_dma_domain.c

```diff
@@ -453,15 +482,20 @@ static int zephyr_dma_domain_unregister(struct ll_schedule_domain *domain,
 	struct zephyr_dma_domain *zephyr_dma_domain;
 	struct zephyr_dma_domain_thread *dt;
 	struct pipeline_task *pipe_task;
+	struct zephyr_dma_domain_data *data;
 	int ret, core;
 
 	zephyr_dma_domain = ll_sch_get_pdata(domain);
 	pipe_task = pipeline_task_get(task);
 	core = cpu_get_id();
 	dt = zephyr_dma_domain->domain_thread + core;
+	data = pipe_task->priv;
 
 	tr_info(&ll_tr, "zephyr_dma_domain_unregister()");
 
+	data->pending_count--;
```

Comment:
In theory this shouldn't happen since `domain_is_pending()` always resets its value when 0. Since cancelling a task is synchronous that means `domain_is_pending()` will always be called.

Reply:
Still need to stress test and if this does happen then we have a really big problem.

Reply:
Yeah I feel like this very decrement can get it below zero, before `domain_is_pending()` has a chance to reset it.

Reply:
Say, if you're somehow stopping two pipelines quickly which leads to unregistering things inbetween them. Say if it's playback and capture (which work with different scheduling component). The following order may cause issues:

1. IRQ arrives for playback; it is handled, pending = 1
2. Playback pipeline is torn down, pending = 0
3. Capture pipeline is torn down before IRQ ever arrives for capture this period, pending = -1

It's pretty hard to actually have this happen I might say, but having "pending" not equate an actual pipeline walk pending simply doesn't sit well with me (variable name hinting one thing while it technically _isn't_ that thing)

Reply:
Okay then my only remaining complaint is that the name can send you down the wrong train of thought. I can accept it if nobody else agrees with me though.

Reply:
maybe add an `assert()` at least